dataframe/dataframe.go

```diff
@@ -7,120 +7,75 @@ import (
 	"time"
 )
 
-// FieldType is used to describe the type of data in a field.
-type FieldType int
-
-// All valid field types.
-const (
-	FieldTypeOther FieldType = iota
-	FieldTypeTime
-	FieldTypeNumber
-	FieldTypeString
-	FieldTypeBoolean
-)
-
-func (f FieldType) String() string {
-	switch f {
-	case FieldTypeOther:
-		return "other"
-	case FieldTypeNumber:
-		return "number"
-	case FieldTypeString:
-		return "string"
-	case FieldTypeBoolean:
-		return "boolean"
-	case FieldTypeTime:
-		return "time"
-	default:
-		return "unknown"
-	}
-}
-
 // Field represents a column of data with a specific type.
 type Field struct {
 	Name   string
-	Type   FieldType
 	Vector Vector
 }
 
+// Fields is a slice of Field pointers.
 type Fields []*Field
 
-func assertFieldType(got, want FieldType) {
-	if got != want {
-		panic(fmt.Sprintf("values doesn't match the specified field type"))
-	}
-}
-
 // NewField returns a new instance of Field.
-func NewField(name string, fieldType FieldType, values interface{}) *Field {
+func NewField(name string, values interface{}) *Field {
 	var vec Vector
-
 	switch v := values.(type) {
 	case []*float64:
-		assertFieldType(fieldType, FieldTypeNumber)
-		vec = newVector(fieldType, len(v))
-		for i := 0; i < len(v); i++ {
-			vec.Set(i, v[i])
-		}
-	case []*int64:
-		assertFieldType(fieldType, FieldTypeNumber)
-		vec = newVector(fieldType, len(v))
+		vec = newVector(v, len(v))
 		for i := 0; i < len(v); i++ {
 			vec.Set(i, v[i])
 		}
+	// case []*int64:
```

Comment:
Any reason this is commented?

Reply:
Just haven't gotten to it yet, will make work before merging.